src/libsyntax/attr.rs

```diff
@@ -557,14 +689,19 @@ fn find_stability_generic<'a, I>(diagnostic: &Handler,
                     let mut since = None;
                     let mut reason = None;
                     for meta in metas {
-                        match &*meta.name() {
-                            "since" => if !get(meta, &mut since) { continue 'outer },
-                            "reason" => if !get(meta, &mut reason) { continue 'outer },
-                            _ => {
-                                handle_errors(diagnostic, meta.span,
-                                              AttrError::UnknownMetaItem(meta.name()));
-                                continue 'outer
+                        if let MetaListItemKind::MetaItem(ref mi) = meta.node {
```

Comment:
Please write a variant of `list_contains_name` (or change its type signature) to return an `Option`, then rewrite this (and maybe also `cfg_matches` to use that instead of the mess that's here right now.